### ISSUE TITLE
logs of `lookup apiserver-svc.default-e7347a-vc-sample-1 failure'

### ISSUE THREAD

**Issue by jichenjc** [kubernetes-sigs/cluster-api-provider-nested#171]
**What steps did you take and what happened:**
[A clear and concise description on how to REPRODUCE the bug.]

**What did you expect to happen:**

follow the VC doc step by step, every pod running well 

```
vc-manager                           vc-manager-76c5878465-bwq2t        1/1     Running   0          21h
vc-manager                           vc-syncer-58fb5d7d49-pbvzp         1/1     Running   0          21h
vc-manager                           vn-agent-p6gqv                     1/1     Running   0          21h


kubectl logs vc-syncer-58fb5d7d49-pbvzp -n vc-manager
```

a lot of such errors in the log
```
E0712 08:17:09.023633       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.21.1/tools/cache/reflector.go:167: Failed to watch *v1.ServiceAccount: failed to list *v1.ServiceAccount: Get "https://[apiserver-svc.default-e7347a-vc-sample-1]:6443/api/v1/serviceaccounts?resourceVersion=240&timeout=30s": dial tcp: lookup apiserver-svc.default-e7347a-vc-sample-1 on 10.96.0.10:53: no such host
E0712 08:17:18.558762       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.21.1/tools/cache/reflector.go:167: Failed to watch *v1.ConfigMap: failed to list *v1.ConfigMap: Get "https://[apiserver-svc.default-e7347a-vc-sample-1]:6443/api/v1/configmaps?resourceVersion=39&timeout=30s": dial tcp: lookup apiserver-svc.default-e7347a-vc-sample-1 on 10.96.0.10:53: no such host
E0712 08:17:27.758210       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.21.1/tools/cache/reflector.go:167: Failed to watch *v1.Pod: failed to list *v1.Pod: Get "https://[apiserver-svc.default-e7347a-vc-sample-1]:6443/api/v1/pods?resourceVersion=1&timeout=30s": dial tcp: lookup apiserver-svc.default-e7347a-vc-sample-1 on 10.96.0.10:53: no such host
E0712 08:17:31.546295       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.21.1/tools/cache/reflector.go:167: Failed to watch *v1.Secret: failed to list *v1.Secret: Get "https://[apiserver-svc.default-e7347a-vc-sample-1]:6443/api/v1/secrets?resourceVersion=239&timeout=30s": dial tcp: lookup apiserver-svc.default-e7347a-vc-sample-1 on 10.96.0.10:53: no such host
E0712 08:17:39.932727       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.21.1/tools/cache/reflector.go:167: Failed to watch *v1.Endpoints: failed to list *v1.Endpoints: Get "https://[apiserver-svc.default-e7347a-vc-sample-1]:6443/api/v1/endpoints?resourceVersion=146&timeout=30s": dial tcp: lookup apiserver-svc.default-e7347a-vc-sample-1 on 10.96.0.10:53: no such host
E0712 08:17:40.618151       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.21.1/tools/cache/reflector.go:167: Failed to watch *v1.Node: failed to list *v1.Node: Get "https://[apiserver-svc.default-e7347a-vc-sample-1]:6443/api/v1/nodes?resourceVersion=1&timeout=30s": dial tcp: lookup apiserver-svc.default-e7347a-vc-sample-1 on 10.96.0.10:53: no such host
E0712 08:17:41.310447       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.21.1/tools/cache/reflector.go:167: Failed to watch *v1.PersistentVolume: failed to list *v1.PersistentVolume: Get "https://[apiserver-svc.default-e7347a-vc-sample-1]:6443/api/v1/persistentvolumes?resourceVersion=1&timeout=30s": dial tcp: lookup apiserver-svc.default-e7347a-vc-sample-1 on 10.96.0.10:53: no such host
E0712 08:17:45.740813       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.21.1/tools/cache/reflector.go:167: Failed to watch *v1.Event: failed to list *v1.Event: Get "https://[apiserver-svc.default-e7347a-vc-sample-1]:6443/api/v1/events?resourceVersion=240&timeout=30s": dial tcp: lookup apiserver-svc.default-e7347a-vc-sample-1 on 10.96.0.10:53: no such host
E0712 08:17:45.789425       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.21.1/tools/cache/reflector.go:167: Failed to watch *v1.Namespace: failed to list *v1.Namespace: Get "https://[apiserver-svc.default-e7347a-vc-sample-1]:6443/api/v1/namespaces?resourceVersion=143&timeout=30s": dial tcp: lookup apiserver-svc.default-e7347a-vc-sample-1 on 10.96.0.10:53: no such host
E0712 08:17:46.682751       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.21.1/tools/cache/reflector.go:167: Failed to watch *v1.Service: failed to list *v1.Service: Get "https://[apiserver-svc.default-e7347a-vc-sample-1]:6443/api/v1/services?resourceVersion=145&timeout=30s": dial tcp: lookup apiserver-svc.default-e7347a-vc-sample-1 on 10.96.0.10:53: no such host
E0712 08:17:47.178369       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.21.1/tools/cache/reflector.go:167: Failed to watch *v1.PersistentVolumeClaim: failed to list *v1.PersistentVolumeClaim: Get "https://[apiserver-svc.default-e7347a-vc-sample-1]:6443/api/v1/persistentvolumeclaims?resourceVersion=1&timeout=30s": dial tcp: lookup apiserver-svc.default-e7347a-vc-sample-1 on 10.96.0.10:53: no such ho

```

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- cluster-api-provider-nested version:
- Minikube/KIND version:
- Kubernetes version: (use `kubectl version`):
- OS (e.g. from `/etc/os-release`):

/kind bug
[One or more /area label. See https://github.com/kubernetes-sigs/cluster-api-provider-nested/labels?q=area for the list of labels]

**Comment by vincent-pli** [comment]
The issue has been fix by #163 
please use the latest image @jichenjc

**Comment by jichenjc** [comment]
thanks for point me out , I am using a latest minikube and pull from docker hub today
how should I get latest one? Thanks

**Comment by vincent-pli** [comment]
it's weird, these issues should be addressed in the latest image, see #165 
@jichenjc

**Comment by jichenjc** [comment]
@vincent-pli thanks for pointing me out, I found the image is 2 weeks ago so I reload the image and the issue seems gone now
